test/nodes/test_other.py

```diff
@@ -56,3 +57,44 @@ def test_joinanswers(join_mode):
     result, _ = join_answers.run(inputs, top_k_join=1)
     assert len(result["answers"]) == 1
     assert result["answers"][0].answer == "answer 2"
+
+
+#
+# JoinDocuments tests
+#
+
+
+@pytest.mark.parametrize("join_mode", ["concatenate", "merge", "reciprocal_rank_fusion"])
+def test_joindocuments(join_mode):
+    inputs = [
+        {"documents": [Document(content="text document 1", content_type="text", score=0.2)]},
+        {"documents": [Document(content="text document 2", content_type="text", score=0.7)]},
+    ]
+
+    join_docs = JoinDocuments(join_mode=join_mode)
+    result, _ = join_docs.run(inputs)
+    assert len(result["documents"]) == 2
+    assert result["documents"] == sorted(result["documents"], reverse=True)
+
+    result, _ = join_docs.run(inputs, top_k_join=1)
+    assert len(result["documents"]) == 1
+    if join_mode == "reciprocal_rank_fusion":
+        assert result["documents"][0].content == "text document 1"
+    else:
+        assert result["documents"][0].content == "text document 2"
+
+
+@pytest.mark.parametrize("join_mode", ["concatenate", "merge", "reciprocal_rank_fusion"])
+def test_joindocuments_score_none(join_mode):
+    """Testing JoinDocuments() node when some of the documents have `score=None`"""
+    inputs = [
+        {"documents": [Document(content="text document 1", content_type="text", score=0.2)]},
+        {"documents": [Document(content="text document 2", content_type="text", score=None)]},
+    ]
+
+    join_docs = JoinDocuments(join_mode=join_mode, sort_by_score=False)
```

Comment:
Let's test `sort_by_score=True` as well

Reply:
Okey, I have extended the test to `sort_by_score=False`, but obviously that will throwing an error when trying to sort by score.
 I could have changed the code to make the sorting work even if the `score=None`, but only be providing a default value to the `score` when it is `None`, which is problematic as not all scores are bounded - so what default value to set is not self-obvious.